File: MindSwipe/Models/Card.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Drawing;

namespace MindSwipe.Models
{
    public class Card
    {
        [Required]
        public int Id {  get; set; }
        [MaxLength(255)]
        public string FrontText {  get; set; }
        [MaxLength(255)]
        public string BackText {  get; set; }
        [MaxLength(255)]
        public string FrontImg {  get; set; }
        [MaxLength(255)]
        public string BackImg { get; set; }
        [Required]
        public DateTime CreationDate { get; set; } = DateTime.Now;
        [Required]
        public DateTime LastModifiedDate { get; set; } = DateTime.Now;
        // Foreign key
        public int? DeckId { get; set; }
        public Deck? Deck { get; set; }
    }
}

File: mind/src/app/models/card.ts
import { Deck } from './deck';

export class Card {
  id: number;
  frontText: string;
  backText: string;
  frontImg: string;
  backImg: string;
  creationDate: Date;
  lastModifiedDate: Date;
  deckId: number;

  constructor(id: number, frontText: string, backText: string, frontImg: string, backImg: string, creationDate: Date, lastModifiedDate: Date, deckId: number) {
      this.id = id;
      this.frontText = frontText;
      this.backText = backText;
      this.frontImg = frontImg;
      this.backImg = backImg;
      this.creationDate = creationDate;
      this.lastModifiedDate = lastModifiedDate;
      this.deckId = deckId;
  }
}
